apis/v1beta1/workspace_types.go

```diff
@@ -135,6 +135,7 @@ type WorkspaceParameters struct {
 
 // WorkspaceObservation are the observable fields of a Workspace.
 type WorkspaceObservation struct {
+	TFPlan   string                       `json:"tfPlan,omitempty"`
```

Comment:
We had better use a pointer type here as the field is optional per [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required) and add the `+optional` marker:
```suggestion
	// +optional 
	TFPlan   *string                       `json:"tfPlan,omitempty"`
```

We may also consider marking this field as required but my understanding is that our `tfclient.Diff` implementation always returns a non-zero value for the plan output (either the string `No Change in terraform plan` or the base64-encoded gzipped output from `terraform plan`). But leaving this field optional sounds better to me.